src/common/helpers/poller.js

```diff
@@ -0,0 +1,272 @@
+  return db.transaction(async (trx) => {
+    const row = await db.model('Repository')
+      .where({ owner, name })
+      .fetch({ transacting: trx });
+    await row.save({ polled_at: _now }, { method: 'update', transacting: trx });
+  });
+};
+
+
+// Whether a given (GitHub) repository has new commits since 'last_polled_at'.
+export const hasRepoChanged = async (repositoryUrl, last_polled_at, token) => {
+  if (last_polled_at === undefined || !last_polled_at) {
+    throw new Error('`last_polled_at` must be given.');
+  }
+  const last_polled = moment(last_polled_at);
+  const since = last_polled.toISOString();
+  const { owner, name } = parseGitHubRepoUrl(repositoryUrl);
+  const uri = `/repos/${owner}/${name}/commits?since=${since}`;
+  const options = {
+    token,
+    headers: {
+      'If-Modified-Since': last_polled.format('ddd, MM MMM YYYY HH:mm:ss [GMT]')
+    },
+    json: true
+  };
+
+  const response = await requestGitHub.get(uri, options);
+
+  switch (response.statusCode) {
+    case 200:
+      // If the (JSON encoded) body is not an empty list.
+      return response.body.length > 0;
+    case 304:
+      // `If-Modified-Since` in action, cache hit, no changes.
+      return false;
+    default:
+      // Bail, unexpected response.
+      throw new Error(
+        `${repositoryUrl} (${response.statusCode}): ${response.body.message}`);
+  }
+};
+
+
+// Extracts unique GH repository URLs from a given (parsed) snapcraft.yaml
+export function extractPartsToPoll(snapcraft_yaml) {
+  const parts = Object.values(snapcraft_yaml.parts || {});
+  const sourceParts = parts.map(
+    GitSourcePart.fromSnapcraftPart).filter(part => part != undefined);
+  return Array.from(new Set(sourceParts));
+}
+
+
+/// GitSourcePart encapsulates the relevant information from a snapcraft
+/// source part, and contains the methods to determine whether it's up to
+/// date.
+export class GitSourcePart {
+
+  constructor(repoUrl, branch, tag) {
+    if (repoUrl === undefined) {
+      throw new Error('Required parameter: repoUrl');
+    }
+    if (branch === undefined) {
+      branch = 'master';
+    }
+    this.repoUrl = repoUrl;
+    this.branch = branch;
+    this.tag = tag;
+  }
+
+  /** Extract a GitSourcePart from a snapcraft part definition.
+   *
+   * Returns a GitSourcePart instance if the source part meets the following
+   * criteria:
+   *
+   * - The source part has a source repository listed, and it's a github
+   *   hosted repository
+   */
+  static fromSnapcraftPart(part) {
+    // TODO: Warn if source-commit or source-subdir are set, since we don't
+    //       support these.
+    // TODO: Not sure if we can support setting tags _and_ branch in the same
+    //       part.
+    const gh_repo_prefix = conf.get('GITHUB_REPOSITORY_PREFIX');
+    if (part.source == undefined) {
+      logger.info('Skipping part with no source set.');
+    } else if (part.source.startsWith(gh_repo_prefix)) {
```

Comment:
GH repo prefix is configured to be "https://github.com" so it's only https repo links that we support, we don't allow to link via "git@" URLs?